typed/merge.go

```diff
@@ -168,7 +219,7 @@ func (w *mergingWalker) visitListItems(t schema.List, lhs, rhs *value.List) (err
 	rhsOrder := []fieldpath.PathElement{}
 
 	// First, collect all RHS children.
-	observedRHS := map[string]value.Value{}
```

Comment:
I'm surprised I missed these uses of strings when I was working on this last quarter...

Reply:
1. We had much bigger things to improve,
2. Maybe because of the test data?!

Anyways, /shrug

Reply:
This PR is a fairly small improvement compared to some of the things that are coming.